javascript-carousel/main.js

```diff
@@ -0,0 +1,181 @@
+  four.classList.remove('background-black');
+  four.classList.add('background-white');
+  five.classList.remove('background-black');
+  five.classList.add('background-white');
+  newInterval = setInterval(countDecrease, 3000);
+  return newInterval;
+});
+three.addEventListener('click', function (event) {
+  clearInterval(newInterval);
+  count = 2;
+  countDecrease();
+  three.classList.remove('background-white');
+  three.classList.add('background-black');
+  two.classList.remove('background-black');
+  two.classList.add('background-white');
+  one.classList.remove('background-black');
+  one.classList.add('background-white');
+  four.classList.remove('background-black');
+  four.classList.add('background-white');
+  five.classList.remove('background-black');
+  five.classList.add('background-white');
+  newInterval = setInterval(countDecrease, 3000);
+  return newInterval;
+});
+four.addEventListener('click', function (event) {
+  clearInterval(newInterval);
+  count = 3;
+  countDecrease();
+  four.classList.remove('background-white');
+  four.classList.add('background-black');
+  two.classList.remove('background-black');
+  two.classList.add('background-white');
+  three.classList.remove('background-black');
+  three.classList.add('background-white');
+  one.classList.remove('background-black');
+  one.classList.add('background-white');
+  five.classList.remove('background-black');
+  five.classList.add('background-white');
+  newInterval = setInterval(countDecrease, 3000);
+  return newInterval;
+});
+five.addEventListener('click', function (event) {
+  clearInterval(newInterval);
+  count = 4;
+  countDecrease();
+  five.classList.remove('background-white');
+  five.classList.add('background-black');
+  two.classList.remove('background-black');
+  two.classList.add('background-white');
+  three.classList.remove('background-black');
+  three.classList.add('background-white');
+  four.classList.remove('background-black');
+  four.classList.add('background-white');
+  one.classList.remove('background-black');
+  one.classList.add('background-white');
+  newInterval = setInterval(countDecrease, 3000);
+  return newInterval;
+});
+
+backward.addEventListener('click', function (event) {
+
+  clearInterval(newInterval);
+  count = count - 2;
+  if (count === 0) {
+    two.classList.add('background-white');
+    two.classList.remove('background-black');
+  } else if (count === 1) {
+    three.classList.add('background-white');
+    three.classList.remove('background-black');
+  } else if (count === 2) {
+    four.classList.add('background-white');
+    four.classList.remove('background-black');
+  } else if (count === 3) {
+    five.classList.add('background-white');
+    five.classList.remove('background-black');
+  }
+  countDecrease();
+  newInterval = setInterval(countDecrease, 3000);
+  return newInterval;
+});
+
+forward.addEventListener('click', function (event) {
+  count++;
+  count = count - 1;
```

Comment:
Don't these two lines just cancel each other out?

Reply:
They do not. The carousel increases each time it is returned. Due to the fact that when that add event listener is clicked, the count would actually increase by two since it is increasing due to the count++ on line 175 and also the count++ inside the countDecrease. I added the count - 1 to ensure it only increases by 1 not 2. I was having that issue and that was the fix I found to work.

Reply:
You're over-thinking it:

---

![image](https://user-images.githubusercontent.com/16143771/172732841-9971dd79-516b-4f87-b534-6fec1ab89db1.png)

---

`count++` is shorthand for `count += 1`, which is shorthand for `count = count + 1`. That turns these two lines into...
```js
count = count + 1;
count = count - 1;
```
... which is the same as ...
```js
count = count + 1 - 1;
```
... which is the same as ...
```js
count = count;
```
... which doesn't change anything.